packages/typespec-ts/src/modular/helpers/operationHelpers.ts

```diff
@@ -633,6 +628,17 @@ export function getResponseMapping(
         )} ${
           !property.optional ? "" : `!${propertyFullName} ? undefined :`
         } ${propertyFullName}`;
+      } else if (
+        (property.restApiName === "message" ||
+          property.restApiName === "messages") &&
+        (property.type.name === "ChatMessage" ||
+          property.type.elementType?.name === "ChatMessage")
+      ) {
```

Comment:
Debugging code?

Reply:
no, this is no easy way to identity whether there's a circular reference with current codegen implementation.

Reply:
Got it, this is just work around to bypass the OpenAI generation.

Reply:
Did you consider doing a breadth-first traversal where we keep track of every type seen so far to detect if the model we are about to dive into has been seen before?

Reply:
This is just a workaround to unblock your openai August release. breadth-first traversal or whatever will be easier to do. after https://github.com/Azure/autorest.typescript/pull/1962 is done.